src/Test/Utilities/Portable/Compilation/TestOperationVisitor.cs

```diff
@@ -597,18 +597,7 @@ public override void VisitEventAssignment(IEventAssignmentOperation operation)
         {
             Assert.Equal(OperationKind.EventAssignment, operation.Kind);
             var adds = operation.Adds;
-            // operation.EventReference shouldn't be null. The following conditional logic is
-            // a work around for https://github.com/dotnet/roslyn/issues/23810 and should 
-            // be removed once the issue is fixed
-            if (operation.EventReference == null)
-            {
-                Assert.Equal(LanguageNames.VisualBasic, operation.Language);
-                Assert.Same(operation.HandlerValue, operation.Children.Single());
-            }
-            else
-            {
-                AssertEx.Equal(new[] { operation.EventReference, operation.HandlerValue }, operation.Children);
-            }
+            AssertEx.Equal(new[] { operation.EventReference, operation.HandlerValue }, operation.Children);
```

Comment:
>operation.EventReference [](start = 35, length = 24)

Consider asserting (Assert.NotNull) that it is not null in OperationTreeVerifier.